CHANGELOG.md

```diff
@@ -3,6 +3,7 @@
 ## Version 0.5.0-dev  (changes since 0.5.0 go here)
 
 ### Features
+* `DissimilarityMatrix` now has a new constructor method called `from_iterable`. ([#1343](https://github.com/biocore/scikit-bio/issues/1343)).
```

Comment:
Can you note that `DissimilarityMatrix` now allows non-hollow matrices, and the addition of `validate=True` parameter to `DistanceMatrix.from_iterable`? Also note in the "backwards-incompatible changes (experimental)" section that `DistanceMatrix.from_iterable` default behavior is now to validate matrix by computing all pairwise distances. Instruct users to pass `validate=False` to get the previous behavior (no validation, but faster execution).